src/iam/roleRequiredFieldsHelper.ts

```diff
@@ -0,0 +1,62 @@
+import { IRoleDefinition } from "@energyweb/iam-contracts";
+import { ERROR_MESSAGES } from "../errors";
+import { buildRegex } from "../utils/regex_builder";
+import Ajv from "ajv";
+export const REQUIRED_PARAMS_FIELD_NAME = "requiredParams";
+const requiredParamsSchema = {
+    type: "object",
+    additionalProperties: true,
+    properties: {
+        [REQUIRED_PARAMS_FIELD_NAME]: {
+            type: "array",
+            uniqueItems: true,
+            items: {
+                type: "object",
+                required: ["name"],
+                properties: {
+                    name: { type: "string" },
+                    pattern: { type: "string" },
+                    description: { type: "string" },
+                },
+            },
+        },
+    },
+};
+
+const ajv = new Ajv();
+const validate = ajv.compile(requiredParamsSchema);
+
+export type RoleRequiredParamSchema = {
+    name: string;
+    pattern?: string;
+    description: string;
+};
+
+export function validateRequiredParamsDefinition(roleDefinition: IRoleDefinition) {
+    if (!(REQUIRED_PARAMS_FIELD_NAME in roleDefinition?.metadata)) return;
+
+    const valid = validate(roleDefinition.metadata);
+    if (!valid) throw new Error(ERROR_MESSAGES.ROLE_REQUIRED_PARAMS_DEFINITION_INVALID);
+}
+
+export function verifyRequiredParams({
+    claimParams,
+    roleDefinition,
+}: {
+    claimParams: Record<string, string> | undefined;
+    roleDefinition: IRoleDefinition;
+}) {
+    if (!roleDefinition) {
+        throw new Error(ERROR_MESSAGES.ROLE_NOT_EXISTS);
+    }
+    if (roleDefinition.metadata && REQUIRED_PARAMS_FIELD_NAME in roleDefinition.metadata) {
```

Comment:
@Passerino I see now that the story details were a little unclear, sorry about that. It is true that we have a `metadata` property in the `IRoleDefinition` but we also use the string "metadata" to index into the ENS Text resolver. https://github.com/energywebfoundation/iam-contracts/blob/3f3b3edcc39645f5b520ed303a33d0368389eb8d/src/DomainReader.ts#L161
I think, therefore, that it is a bit strange that our `fields` implementation would be different from our `requiredParams` implementation. We currently have our `fields` as a "first-class" property of the `IRoleDefinition` and I think these `issuerFields` should be too. I can see how you ended up putting them in the existing "metadata" of the `IRoleDefinition` but I think we should have another `issuerFields` property of `IRoleDefinition`